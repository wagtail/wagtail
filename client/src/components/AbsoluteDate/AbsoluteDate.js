import React from 'react';
import moment from 'moment';

import { DATE_FORMAT, STRINGS } from '../../config/wagtail';

const AbsoluteDate = ({ time }) => {
  const date = moment(time);
  const text = time ?  date.format(DATE_FORMAT) : STRINGS.NO_DATE;

  return (
    <span>{text}</span>
  );
};

AbsoluteDate.propTypes = {
  time: React.PropTypes.string,
};

AbsoluteDate.defaultProps = {
  time: '',
};

export default AbsoluteDate;
