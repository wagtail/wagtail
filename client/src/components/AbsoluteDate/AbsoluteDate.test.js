import React from 'react';
import { shallow } from 'enzyme';

import AbsoluteDate from './AbsoluteDate';

describe('AbsoluteDate', () => {
  it('exists', () => {
    expect(AbsoluteDate).toBeDefined();
  });

  it('basic', () => {
    expect(shallow(<AbsoluteDate />)).toMatchSnapshot();
  });

  it('#time', () => {
    expect(shallow(<AbsoluteDate time="2016-09-19T20:22:33.356623Z" />)).toMatchSnapshot();
  });
});
