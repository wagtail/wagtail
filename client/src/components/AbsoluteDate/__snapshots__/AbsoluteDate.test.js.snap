exports[`AbsoluteDate #time 1`] = `
<span>
  19.09.2016
</span>
`;

exports[`AbsoluteDate basic 1`] = `
<span>
  No date
</span>
`;
